rest-core/src/Rest/Driver/Routing.hs

```diff
@@ -319,14 +319,15 @@ mkMultiHandler sBy run (GenHandler dict act sec) = GenHandler <$> mNewDict <*> p
               mapExceptT (run i) (act (Env hs ps v))
          return . StringHashMap.fromList $ zipWith (\(k, _) b -> (k, eitherToStatus b)) (StringHashMap.toList vs) bs
 
-mkMultiGetHandler :: forall m s. (Applicative m, Monad m) => Rest.Router m s -> Handler m
-mkMultiGetHandler root = mkInputHandler (xmlJsonI . multipartO) $ \(Resources rs) -> multiGetHandler rs
+mkMultiGetHandler :: forall m s. (Applicative m, Monad m) => Config m -> Rest.Router m s -> Handler m
+mkMultiGetHandler cfg root = mkInputHandler (xmlJsonI . multipartO) $ \(Resources rs) -> multiGetHandler rs
   where
-    multiGetHandler rs = lift $
-      do mapM (runResource root) rs
+    multiGetHandler rs = lift . forM rs $ \r -> do
+      o <- (multiGetHook cfg) r (runResource cfg root r)
```

Comment:
I'm also not sure. Do you see any practical case where one would like to replace `runResource` by something else? Otherwise, using it just as hook would make purpose of this feature more clear.